src/sinks/opendal_common.rs

```diff
@@ -0,0 +1,218 @@
+//! opendal_common provide real sink supports for all opendal based services.
+//!
+//! # TODO
+//!
+//! opendal service now only support very basic sink features. To make it
+//! useful, we need to add the following features:
+//!
+//! - Batch events
+//! - Error handling
+//! - Limitation
+//! - Compression
+//! - KeyPartition
```

Comment:
**thought (non-blocking)**: It seems like it would be straightforward to provide a simple partitioning mechanism in the `split_input()` function, but the sink would be usable without it.